.github/workflows/plugin-check.yml

```
@@ -0,0 +1,71 @@
+name: Plugin Check
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - develop
+            - trunk
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - ready_for_review
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+    # The concurrency group contains the workflow name and the branch name for pull requests
+    # or the commit hash for any other events.
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+# Disable permissions for all available scopes by default.
+# Any needed permissions should be configured at the job level.
+permissions: {}
+
+jobs:
+    plugin-check:
+        name: Run Plugin Check
+        runs-on: 'ubuntu-24.04'
+        permissions:
+            contents: read
+        timeout-minutes: 20
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+              with:
+                  show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+                  persist-credentials: false
+
+            - name: Set up PHP
+              uses: shivammathur/setup-php@ec406be512d7077f68eed36e63f4d91bc006edc4 # v2.35.4
+              with:
+                  php-version: '8.3'
+                  coverage: none
+
+            - name: Install Composer dependencies
+              uses: ramsey/composer-install@3cf229dc2919194e9e36783941438d17239e8520 # v3.1.1
+
+            - name: Setup Node
+              uses: actions/setup-node@a0853c24544627f65ddf259abe73b1d18a591444 # v5.0.0
+              with:
+                  cache: 'npm'
+                  node-version-file: '.nvmrc'
+
+            - name: Install NPM dependencies
+              run: npm ci
+
+            - name: Build assets
+              run: npm run build
+
+            - name: Run plugin check
+              uses: wordpress/plugin-check-action@v1
```

https://github.com/WordPress/plugin-check-action/commit/ec9b3fe9beaa76bcc4510b7ba2cb5855a5f80f3f

```suggestion
              uses: wordpress/plugin-check-action@ec9b3fe9beaa76bcc4510b7ba2cb5855a5f80f3f # v1.1.4
```